### PR TITLE
ENH: mul(Tick, float); simplify to_offset

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

This will let us remove Resolution.get_stride_from_decimal

### CODE REVIEW

File: pandas/_libs/tslibs/offsets.pyx [hunk 1]
```diff
@@ -743,6 +747,25 @@ cdef class Tick(SingleConstructorOffset):
                 "Tick offset with `normalize=True` are not allowed."
             )
 
+    # FIXME: Without making this cpdef, we get AttributeError when calling
+    #  from __mul__
+    cpdef Tick _next_higher_resolution(Tick self):
```
Comment (jreback) [inline pandas/_libs/tslibs/offsets.pyx:752]: wouldn't this be better as a module level routine?

File: pandas/_libs/tslibs/offsets.pyx [hunk 2]
```diff
@@ -3563,6 +3601,9 @@ cpdef to_offset(freq):
     >>> to_offset(Hour())
     <Hour>
     """
+    # TODO: avoid runtime imports
```
Comment (jreback) [inline pandas/_libs/tslibs/offsets.pyx:3604]: really really prefer absolute imports

File: pandas/_libs/tslibs/offsets.pyx [hunk 3]
```diff
@@ -3598,15 +3641,19 @@ cpdef to_offset(freq):
                 if not stride:
                     stride = 1
```
Comment (jreback) [inline pandas/_libs/tslibs/offsets.pyx:3643]: can you give some comments on what is being done here

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/tseries/offsets/test_ticks.py [hunk 4]
```diff
@@ -244,6 +244,22 @@ def test_tick_division(cls):
...
+    off = Micro(2)
+
+    # Case where we retain type
+    result = off * 1.5
+    expected = Micro(3)
+    assert result == expected
+    assert isinstance(result, Micro)
+
+    # Case where we bump up to the next type
+    result = off * 1.25
```
Comment (jreback) [inline pandas/tests/tseries/offsets/test_ticks.py:257]: was this buggy before? need a whatsnew note if so

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_libs/tslibs/offsets.pyx (hunk 2 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/offsets.pyx:3604]: sure.  im more concerned about getting rid of the runtime imports

File: pandas/tests/tseries/offsets/test_ticks.py (hunk 4 above)
Comment (jbrockmendel) [reply pandas/tests/tseries/offsets/test_ticks.py:257]: ATM it raises TypeError for any float, will add whatsnew

File: pandas/_libs/tslibs/offsets.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/offsets.pyx:752]: i dont think so.  it uses `self.n`, so really makes sense as a method

**Comment by jreback** [comment]
kk if u can rebase

**Comment by jbrockmendel** [comment]
rebased+green

**Review by jreback — APPROVED** [review]

**Pull request merged**